src/resource/task/task.service.ts

```diff
@@ -1,47 +1,59 @@
-import { Injectable, NotFoundException } from '@nestjs/common';
+import { Injectable } from '@nestjs/common';
 import { InjectRepository } from '@nestjs/typeorm';
 import { TaskEntity } from 'src/database/entities/task.entity';
 import { Repository } from 'typeorm';
 import { EditTaskDTO, ReturnTaskDTO, CreateTaskDTO } from './dto';
+import { UserEntity } from 'src/database/entities/user.entity';
 
 @Injectable()
 export class TaskService {
     constructor(
         @InjectRepository(TaskEntity)
         private readonly taskRepository: Repository<TaskEntity>,
     ) {}
-
-    async editTask(taskId: number,userId: number, editTask: EditTaskDTO): Promise<ReturnTaskDTO> {
-        const task = await this.taskRepository.findOneBy({ id: taskId,userId: userId });
+    async editTask(
+        user: UserEntity,
+        id: number,
+        editTask: EditTaskDTO,
+    ): Promise<ReturnTaskDTO> {
+        const task = await this.taskRepository.findOneBy({
+            id,
+            user,
+        });
         if (!task) {
             return null;
         }
-        await this.taskRepository.update({ id: taskId ,userId: userId}, { ...editTask });
-        return await this.taskRepository.findOneBy({ id: taskId,userId: userId });;
+        return await this.taskRepository.save({ id, user, ...editTask });
     }
-    
-
-    async getTask(userId:number,taskId: number): Promise<ReturnTaskDTO> {
-        return await this.taskRepository.findOneBy({ userId:userId,id: taskId });
 
+    async getTask(user: UserEntity, id: number): Promise<ReturnTaskDTO> {
+        return this.taskRepository.findOneBy({ id, user });
     }
 
-    async getAllTasks(userId: number): Promise<ReturnTaskDTO[]> {
-        return await this.taskRepository.find({ where: { userId: userId } });
+    async getAllUserTasks(user: UserEntity): Promise<ReturnTaskDTO[]> {
+        return await this.taskRepository.find({
+            relations: ['user'],
+            where: { user },
+        });
     }
 
-    async addTask(userId: number,createTaskDTO: CreateTaskDTO): Promise<ReturnTaskDTO> {
+    async addTask(
+        user: UserEntity,
+        createTaskDTO: CreateTaskDTO,
+    ): Promise<ReturnTaskDTO> {
         const newTask = this.taskRepository.create(createTaskDTO);
-        newTask.userId = userId;
-        return await this.taskRepository.save(newTask);
+        return await this.taskRepository.save({ ...newTask, user });
     }
 
-    async deleteTask(userId: number,taskId: number): Promise<boolean> {
-        const task = await this.taskRepository.findOneBy({ id: taskId, userId: userId });
-        if (!task) {
-            return false;
+    async deleteTask(user: UserEntity, id: number): Promise<boolean> {
+        const task = await this.taskRepository.findOneBy({
+            id,
+            user,
+        });
+        if (task) {
+            await this.taskRepository.remove(task);
```

Comment:
If this function will throw error on fail then it should be ok, otherwise it has to be changed